tpmutil/run_windows.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tpmutil
+
+import (
+	"fmt"
+	"io"
+	"syscall"
+	"unsafe"
+)
+
+// Tbs.dll Docs:
+// https://docs.microsoft.com/en-us/windows/desktop/TBS/tpm-base-services-portal
+var tbsDll = syscall.NewLazyDLL("Tbs.dll")
+var tbsCreateContext = tbsDll.NewProc("Tbsi_Context_Create")
+var tbsSubmitCommand = tbsDll.NewProc("Tbsip_Submit_Command")
+var tbsContextClose = tbsDll.NewProc("Tbsip_Context_Close")
+
+// tbsContextParams2 Docs:
+// https://docs.microsoft.com/en-us/windows/desktop/api/Tbs/ns-tbs-tdtbs_context_params2
+type tbsContextParams2 struct {
+	version uint32
+	flags   uint32
+}
+
+const (
```

Comment:
Where do these come from?

Reply:
Mostly the tbs.h file. Microsoft does not have it on a documentation page, but here's the header file on github.